### PR TITLE
PIV-296 - Update Auth Docs

### PR DESCRIPTION
**Pull request by beaugrantham-va**
https://vasdvp.atlassian.net/browse/PIV-296
https://vasdvp.atlassian.net/browse/PIV-254

Update auth docs with the following:
* PKCE support
* Example showing `launch/patient` scope

### CODE REVIEW

File: src/content/apiDocs/oauthTechnical.mdx [hunk 1]
```diff
@@ -13,79 +13,181 @@ While we test our OpenID Connect flow to ensure the best experience for Veterans
...
+After being approved to use VA OpenID Connect, you'll receive a *client id* and, if you're building a server-based application, a *client secret*.
+
+We support [Authorization Code Flow](#authorization-code-flow) with a *client secret* and [Proof Key for Code Exchange](#proof-key-for-code-exchange) (PKCE) for public apps.
 
 We use separate authorization servers for our development and production environments. This documentation uses the development environment authorization server URL. You'll be provided with the production authorization server URL when you are approved for production access.
 
-### Initiating Authorization
+### Authorization Code Flow
+
+This flow is used by applications that are able to safely store a *client secret* such as a server-based applications.
```
Comment (kalilsn) [inline src/content/apiDocs/oauthTechnical.mdx:24]:
```suggestion
This flow is used by applications that are able to safely store a *client secret* such as server-based applications.
```

File: src/content/apiDocs/oauthTechnical.mdx [hunk 2]
```diff
@@ -13,79 +13,181 @@ While we test our OpenID Connect flow to ensure the best experience for Veterans
...
+Authorization: Basic {base64 encoded *client id* + ':' + *client secret*}
+
+grant_type=refresh_token&refresh_token={your refresh_token}&scope={space separated scopes}
+```
+
+You will receive the same kind of response as above and be able to use your new `access_token` normally.
+
+### Proof Key for Code Exchange
+
+The PKCE flow ([RFC 7636](https://tools.ietf.org/html/rfc7636)) is used by public applications that are unable to safely store a *client secret* such as a navite mobile app.
```
Comment (kalilsn) [inline src/content/apiDocs/oauthTechnical.mdx:131]:
```suggestion
The PKCE flow ([RFC 7636](https://tools.ietf.org/html/rfc7636)) is used by public applications that are unable to safely store a *client secret* such as a native mobile app.
```

**Comment by beaugrantham-va** [comment]
> Glad to see this so clearly documented! Unfortunately the sidebar nav isn't actually generated dynamically, so changes to the headers in this markdown should be accompanied by changes in https://github.com/department-of-veterans-affairs/developer-portal/blob/master/src/containers/documentation/DocumentationRoot.tsx#L50-L56
> 
> For example `#sample-application` now needs to be `#sample-applications`

Thanks; never would have caught this on my own!

**Comment by beaugrantham-va** [comment]
This is ready to for prod at your convenience @department-of-veterans-affairs/lighthouse-interrobang given that #335 is live.

**Pull request merged**